lib/libc/common/source/time/asctime_r.c

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2024 Meta Platforms
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <time.h>
+
+#include <zephyr/sys/util.h>
+
+char *asctime_r(const struct tm *tp, char *buf)
+{
+	static const char wday_str[7][3] = {
+		"Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat",
+	};
+	static const char mon_str[12][3] = {
+		"Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec",
+	};
+	char tmp[128];
```

Comment:
Why have 128 bytes long tmp buffer if we only copy 26 bytes from it?

Reply:
It has to be >> 26, otherwise there will be a build warning from the strcpy function below

Reply:
```
/home/ycsin/zephyrproject/zephyr/lib/libc/common/source/time/asctime_r.c: In function 'asctime_r':
/home/ycsin/zephyrproject/zephyr/lib/libc/common/source/time/asctime_r.c:28:65: warning: '%d' directive output may be truncated writing between 1 and 11 bytes into a region of size between 0 and 16 [-Wformat-truncation=]
   28 |         snprintf(tmp, sizeof(tmp), "%.3s %.3s%3d %.2d:%.2d:%.2d %d\n", wday_str[tp->tm_wday],
      |                                                                 ^~
/home/ycsin/zephyrproject/zephyr/lib/libc/common/source/time/asctime_r.c:28:36: note: directive argument in the range [-2147481748, 2147483647]
   28 |         snprintf(tmp, sizeof(tmp), "%.3s %.3s%3d %.2d:%.2d:%.2d %d\n", wday_str[tp->tm_wday],
      |                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/ycsin/zephyrproject/zephyr/lib/libc/common/source/time/asctime_r.c:28:9: note: 'snprintf' output between 17 and 68 bytes into a destination of size 30
   28 |         snprintf(tmp, sizeof(tmp), "%.3s %.3s%3d %.2d:%.2d:%.2d %d\n", wday_str[tp->tm_wday],
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   29 |                  mon_str[tp->tm_mon], tp->tm_mday, tp->tm_hour, tp->tm_min, tp->tm_sec,
      |                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   30 |                  1900 + tp->tm_year);
      |                  ~~~~~~~~~~~~~~~~~~~
```